cmd/oceanbench/set.go

```diff
@@ -374,6 +376,99 @@ func editDevicesHandler(w http.ResponseWriter, r *http.Request) {
 	http.Redirect(w, r, "/set/devices?ma="+ma, http.StatusFound)
 }
 
+// configDevicesHandler handles configuration of new devices.
+//
+// Form Fields:
+//
+//	name = Name of the new device
+//	ma = MAC address
+//	dt = device type
+//	ssid = WiFi name (optional)
+//	ssid = WiFi password (optional)
+//	lat = latitude (optional)
+//	long = longitude (optional)
+func configDevicesHandler(w http.ResponseWriter, r *http.Request) {
+	logRequest(r)
+	ctx := context.Background()
+	profile, err := getProfile(w, r)
+	if err != nil {
+		if err != gauth.TokenNotFound {
+			log.Printf("authentication error: %v", err)
+		}
+		http.Redirect(w, r, "/", http.StatusUnauthorized)
+		return
+	}
+	skey, _ := profileData(profile)
+
+	// Parse the form values.
+	name := r.FormValue("name")
+	ma := r.FormValue("ma")
+	dt := r.FormValue("dt")
+	ssid := r.FormValue("ssid")
+	pass := r.FormValue("pass")
+	sLat := r.FormValue("lat")
+	sLong := r.FormValue("long")
+	r.ParseForm()
+
+	var lat, long float64
+	if sLat != "" && sLong != "" {
+		lat, err = strconv.ParseFloat(sLat, 64)
+		if err != nil {
+			writeError(w, fmt.Errorf("unable to parse float64 from: %s, err:", err))
```

Comment:
Looks like you were intending to include the variable in the log message but instead you're including the error in the wrong spot.

Reply:
done